### PR TITLE
#166790053 user should be able to: create, update, get and delete article

### PR DESCRIPTION
**Pull request by pngabo**
#### What does this PR do?
  - Implement all crud for article
#### Description
 - Design article's page
 - consume create article API
 - consume update article API
 - consume get one article API
 - consume get all articles API
 - consume delete article API
 - write tests for all crud
#### How to test this PR
 - clone the repo => cd into it => npm install => npm run test =>npm run dev
 #### Relevant pivotaltracker story
 [#166790053](https://www.pivotaltracker.com/story/show/166790053)

### CODE REVIEW

File: src/__tests__/components/article/homePage.test.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    // const props = {props}
+    const wrapper = mount(
+        <MemoryRouter>
+            <Provider store={store}><Home {...props}/></Provider>
+        </MemoryRouter>
+    );
+    wrapper.setProps({props})
+    expect(wrapper).toHaveLength(1);
+  });
+});
```
Comment (mnzube) [inline src/__tests__/components/article/homePage.test.js]: Hey Patrick, add a newline here

File: src/__tests__/components/article/readArticle.test.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+describe('<ReadArticle/>', () => {
+  it('Should render', () => {
+    const wrapper = mount(
+    <MemoryRouter>
+      <Provider store={store}><ReadArticle {...props}/></Provider>
+    </MemoryRouter>
+    );
+    expect(wrapper.contains(<div></div>));
+  });
+});
```
Comment (mnzube) [inline src/__tests__/components/article/readArticle.test.js]: here too

File: src/__tests__/components/article/homePage.test.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    "title": "this is the title",
+    "body": "this is the body of teh article",
+    "author": {
+        "username": "Joe",
+    }  }]}
+
+
+describe('<Home />', () => {
+  it('Should render', () => {
+    // const props = {props}
```
Comment (billkabanga) [inline src/__tests__/components/article/homePage.test.js]: Please remove this commented code

**Review by kagabof — COMMENTED** [review]
Add a gif image to show UI of this PR.

**Review by shaluchandwani — DISMISSED** [review]
While reviewing I found out some issues, Please address them - 
1. The home page is not responsive.
2. When we create an article it redirects me to the home page and then when I open profile page to see that article, the CSS is breaking.

File: src/__tests__/components/article/homePage.test.js (hunk 3 above)
Comment (shaluchandwani) [inline src/__tests__/components/article/homePage.test.js]: remove the commented lines please.

Comment (pngabo) [reply src/__tests__/components/article/homePage.test.js]: @shaluchandwani I remove it

**Review by shaluchandwani — APPROVED** [review]
Great work on the PR @patrickngabo  @Joe-Joseph.

**Review by mnzube — APPROVED** [review]

File: src/components/article/ArticleComponent.js [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+  handleSubmit = async (e) => {
+    e.preventDefault();
+    await this.props.postDataThunkPrivate('post', '/articles', articleAction, this.state);
+    // eslint-disable-next-line no-unused-expressions
+    this.props.article.article.errors ? (
+      toast.error(this.props.article.article.errors.title),
+      toast.error(this.props.article.article.errors.body))
+      : (
+        // this.props.history.push('/'),
```
Comment (billkabanga) [inline src/components/article/ArticleComponent.js]: Please remove this line.

**Pull request merged**